lib/networkaccessmanager.cpp

```diff
@@ -15,7 +15,12 @@ class NetworkAccessManager::Private {
 
 NetworkAccessManager::NetworkAccessManager(QObject* parent)
     : QNetworkAccessManager(parent), d(std::make_unique<Private>())
-{}
+{
+    connect(this, &QNetworkAccessManager::sslErrors, this, [](QNetworkReply *reply, const QList<QSslError> &errors)
```

Comment:
Is this related?

Reply:
it's for the test but need a better solution because we can't ignore the SSL errors in production...

Reply:
Patch that should fix it but doesn't work. Probably because of a stupid error somewhere.

```
diff --git a/autotests/testolmaccount.cpp b/autotests/testolmaccount.cpp
index 9134224..eb44791 100644
--- a/autotests/testolmaccount.cpp
+++ b/autotests/testolmaccount.cpp
@@ -165,6 +165,7 @@ void TestOlmAccount::encryptedFile()
 
 #define CREATE_CONNECTION(VAR, USERNAME, SECRET, DEVICE_NAME) \
     auto VAR = std::make_shared<Connection>(); \
+    VAR->ignoreSslErrors(true); \
     (VAR) ->resolveServer("@alice:localhost:" + QString::number(443)); \
     connect( (VAR) .get(), &Connection::loginFlowsChanged, this, [this, VAR ] () { \
         (VAR) ->loginWithPassword( (USERNAME) , SECRET , DEVICE_NAME , ""); \
diff --git a/lib/connection.cpp b/lib/connection.cpp
index 070aac0..d01aab3 100644
--- a/lib/connection.cpp
+++ b/lib/connection.cpp
@@ -573,6 +573,11 @@ void Connection::sync(int timeout)
     });
 }
 
+void Connection::ignoreSslErrors(bool ignore)
+{
+    connectionData()->ignoreSslErrors(ignore);
+}
+
 void Connection::syncLoop(int timeout)
 {
     if (d->syncLoopConnection && d->syncTimeout == timeout) {
diff --git a/lib/connection.h b/lib/connection.h
index 9a952e0..f3a6b8c 100644
--- a/lib/connection.h
+++ b/lib/connection.h
@@ -481,6 +481,11 @@ public:
         setUserFactory(defaultUserFactory<T>());
     }
 
+    /// Ignore ssl errors (usefull for automated testing with local synapse
+    /// instance).
+    /// \internal
+    void ignoreSslErrors(bool ignore);
+
 public Q_SLOTS:
     /** Set the homeserver base URL */
     void setHomeserver(const QUrl& baseUrl);
diff --git a/lib/connectiondata.cpp b/lib/connectiondata.cpp
index e54d909..123febe 100644
--- a/lib/connectiondata.cpp
+++ b/lib/connectiondata.cpp
@@ -140,6 +140,12 @@ bool ConnectionData::needsToken(const QString& requestName) const
            != d->needToken.cend();
 }
 
+void ConnectionData::ignoreSslErrors(bool ignore) const
+{
+    auto quotientNam = static_cast<NetworkAccessManager>(nam());
+    quotientNam.ignoreSslErrors(ignore);
+}
+
 void ConnectionData::setDeviceId(const QString& deviceId)
 {
     d->deviceId = deviceId;
diff --git a/lib/connectiondata.h b/lib/connectiondata.h
index 7dd96f2..00d056e 100644
--- a/lib/connectiondata.h
+++ b/lib/connectiondata.h
@@ -29,6 +29,7 @@ public:
     bool needsToken(const QString& requestName) const;
     QNetworkAccessManager* nam() const;
 
+    void ignoreSslErrors(bool ignore = true) const;
     void setBaseUrl(QUrl baseUrl);
     void setToken(QByteArray accessToken);
     [[deprecated("Use setBaseUrl() instead")]]
diff --git a/lib/networkaccessmanager.cpp b/lib/networkaccessmanager.cpp
index c6e761e..1276305 100644
--- a/lib/networkaccessmanager.cpp
+++ b/lib/networkaccessmanager.cpp
@@ -16,10 +16,6 @@ public:
 NetworkAccessManager::NetworkAccessManager(QObject* parent)
     : QNetworkAccessManager(parent), d(std::make_unique<Private>())
 {
-    connect(this, &QNetworkAccessManager::sslErrors, this, [](QNetworkReply *reply, const QList<QSslError> &errors)
-            {
-                reply->ignoreSslErrors();
-            });
 }
 
 QList<QSslError> NetworkAccessManager::ignoredSslErrors() const
@@ -27,6 +23,19 @@ QList<QSslError> NetworkAccessManager::ignoredSslErrors() const
     return d->ignoredSslErrors;
 }
 
+void NetworkAccessManager::ignoreSslErrors(bool ignore) const
+{
+    if (ignore) {
+        connect(this, &QNetworkAccessManager::sslErrors, this, [](QNetworkReply *reply, const QList<QSslError> &errors)
+                {
+    qDebug() << 24 / 0;
+                    reply->ignoreSslErrors();
+                });
+    } else {
+        disconnect(this, &QNetworkAccessManager::sslErrors, this, nullptr);
+    }
+}
+
 void NetworkAccessManager::addIgnoredSslError(const QSslError& error)
 {
     d->ignoredSslErrors << error;
diff --git a/lib/networkaccessmanager.h b/lib/networkaccessmanager.h
index 47729a1..22280e0 100644
--- a/lib/networkaccessmanager.h
+++ b/lib/networkaccessmanager.h
@@ -17,6 +17,7 @@ public:
     QList<QSslError> ignoredSslErrors() const;
     void addIgnoredSslError(const QSslError& error);
     void clearIgnoredSslErrors();
+    void ignoreSslErrors(bool ignore = true) const;
 
     /** Get a pointer to the singleton */
     static NetworkAccessManager* instance();
```